src/simulator/Simulator.tsx

```diff
@@ -39,6 +39,15 @@ const Simulator = ({ simulatorShown, setSimulatorShown }: SimulatorProps) => {
   const hideSimulator = useCallback(() => {
     setSimulatorShown(false);
   }, [setSimulatorShown]);
+  useEffect(() => {
+    if (simulatorShown) {
+      ref.current!.focus();
+    } else {
+      (
+        document.querySelector("#simulator-expand") as HTMLButtonElement
```

Comment:
Not sure I like the action-at-a-distance nature of this. Wondering how it would be with a ref.

Reply:
Switch to using a ref here -> [`3e08dc7`](https://github.com/microbit-foundation/python-editor-next/pull/873/commits/3e08dc7ba7c3b89cb54aa6098236f9dbd7e02682). It adds a little noise, but it's not as bad as I thought it would be. Everything will be a lot cleaner once the simulator flag is no longer needed.